crest/Assets/Crest/Crest/Scripts/OceanRenderer.cs

```diff
@@ -1011,38 +1015,61 @@ void LateUpdateLods()
 
         void LateUpdateTiles()
         {
-            // If there are local bodies of water, this will do overlap tests between the ocean tiles
-            // and the water bodies and turn off any that don't overlap.
-            if (WaterBody.WaterBodies.Count == 0 && _canSkipCulling)
+            var isUnderwaterActive = UnderwaterRenderer.Instance != null;
+
+            var definitelyUnderwater = false;
+            var volumeExtinctionLength = 0f;
+
+            if (isUnderwaterActive)
             {
-                return;
+                definitelyUnderwater = ViewerHeightAboveWater < -5f;
+                var density = _material.GetVector("_DepthFogDensity");
+                var minimumFogDensity = Mathf.Min(Mathf.Min(density.x, density.y), density.z);
+                volumeExtinctionLength = -Mathf.Log(_underwaterCullLimit) / minimumFogDensity;
```

Comment:
out of paranoia i think i'd like us to bound _underwaterCullLimit from below. i know it already has a Range() on the param, but perhaps it gets 0 due to some dodgy serialisation scenario, or maybe someone sets it from code at runtime.

Reply:
Done. I used clamp to enforce minimum and maximum.

Reply:
great